Failure on aarch64-linux [(full log)](https://logs.nix.ci/?key=nixos/nixpkgs.41453&attempt_id=d10465b6-fae6-49a7-8d4f-aef87850095d)

Attempted: python36Packages.cython

<details><summary>Partial log (click to expand)</summary><p>

```
                 from /nix/store/sh7z6vib9kxhsj5rjydcd0hqs9acs5y0-python3.6-numpy-1.14.3/lib/python3.6/site-packages/numpy/core/include/numpy/ndarrayobject.h:18,
                 from /nix/store/sh7z6vib9kxhsj5rjydcd0hqs9acs5y0-python3.6-numpy-1.14.3/lib/python3.6/site-packages/numpy/core/include/numpy/arrayobject.h:4,
                 from memoryviewattrs.cpp:580:
/nix/store/sh7z6vib9kxhsj5rjydcd0hqs9acs5y0-python3.6-numpy-1.14.3/lib/python3.6/site-packages/numpy/core/include/numpy/npy_1_7_deprecated_api.h:15:2: warning: #warning "Using deprecated NumPy API, disable it by " "#defining NPY_NO_DEPRECATED_API NPY_1_7_API_VERSION" [-Wcpp]
 #warning "Using deprecated NumPy API, disable it by " \
  ^~~~~~~
acquired default
acquired Global_A
building of '/nix/store/7g7hv0caz1wr66ca0r4yj2v867mnghd6-python3.6-Cython-0.28.2.drv' timed out after 3600 seconds
error: build of '/nix/store/7g7hv0caz1wr66ca0r4yj2v867mnghd6-python3.6-Cython-0.28.2.drv' failed
```
</p></details>

--- a/pkgs/development/python-modules/Cython/default.nix
+++ b/pkgs/development/python-modules/Cython/default.nix
@@ -11,7 +11,18 @@
 , ncurses
 }:
 
-buildPythonPackage rec {
+let
+  excludedTests = []
+    # cython's testsuite is not working very well with libc++
+    # We are however optimistic about things outside of testsuite still working
+    ++ stdenv.lib.optionals (stdenv.cc.isClang or false) [ "cpdef_extern_func" "libcpp_algo" ]
+    # Some tests in the test suite isn't working on aarch64. Disable them for
+    # now until upstream finds a workaround.
+    # Upstream issue here: https://github.com/cython/cython/issues/2308
+    ++ stdenv.lib.optionals stdenv.isAarch64 [ "numpy_memoryview" ]
+  ;
+
+in buildPythonPackage rec {
   pname = "Cython";
   version = "0.28.2";
 
@@ -29,12 +40,11 @@ buildPythonPackage rec {
   buildInputs = [ glibcLocales gdb ];
   LC_ALL = "en_US.UTF-8";
 
-  # cython's testsuite is not working very well with libc++
-  # We are however optimistic about things outside of testsuite still working
   checkPhase = ''
     export HOME="$NIX_BUILD_TOP"
     ${python.interpreter} runtests.py \
-      ${if stdenv.cc.isClang or false then ''--exclude="(cpdef_extern_func|libcpp_algo)"'' else ""}
+      ${stdenv.lib.optionalString (builtins.length excludedTests != 0)
+        ''--exclude="(${builtins.concatStringsSep "|" excludedTests})"''}
   '';
 
   meta = {

